packages/validator/src/services/indices.ts

```diff
@@ -71,7 +85,42 @@ export class IndicesService {
         newIndices.push(validatorState.index);
       }
     }
-
     return newIndices;
+  };
+}
+
+type Batch = string[][];
+const PUBKEYS_PER_REQUEST = 10;
+const REQUESTS_PER_BATCH = 5;
+const MAX_PUBKEYS_PER_POLL = 5 * PUBKEYS_PER_REQUEST * REQUESTS_PER_BATCH;
+
+/**
+ * Divide pubkeys into batches, each batch contains at most 5 http requests,
+ * each request can work on at most 40 pubkeys.
+ * @param pubkeysHex
+ */
+export function pubkeysToBatches(
+  pubkeysHex: string[],
+  maxPubkeysPerRequest: number = PUBKEYS_PER_REQUEST,
+  maxRequesPerBatch: number = REQUESTS_PER_BATCH
+): Batch[] {
+  if (!pubkeysHex || pubkeysHex.length === 0) {
+    return [[[]]];
```

Comment:
Why nesting three levels deep? We run a Promise.all() inside a Promise.all() so it should be equivalent to having a `[[]]` data structure + a single Promise.all()

Reply:
I intended to make a `for loop` instead but missed that. So the intention is:
+ Given 10 validators per http request,  with 2000 validators we'll have to run 200 http requests
+ So I divided into `x` batches, each batch contains `y` http requests and run each batch one by one. We can tweak number of requests per batch to a suitable number

I was afraid running all http requests at the same time is not a scalable solution if we increase the validators, do you think so?

Reply:
> I was afraid running all http requests at the same time is not a scalable solution if we increase the validators, do you think so?

Yeah I agree but aren't we just doing all the HTTP calls at the same time?

Reply:
I should change the 1st `Promise.all` as a `for loop` instead